crates/cairo-lint-core/src/fix.rs

```diff
@@ -232,4 +235,55 @@ impl Fixer {
             None
         }
     }
+
+    pub fn fix_double_comparison(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
+        let expr = Expr::from_syntax_node(db, node.clone());
+
+        if let Expr::Binary(binary_op) = expr {
+            let lhs = binary_op.lhs(db);
+            let rhs = binary_op.rhs(db);
+
+            if let (Expr::Binary(lhs_inner), Expr::Binary(rhs_inner)) = (&lhs, &rhs) {
+                let lhs_var = Self::extract_variable(lhs_inner, db);
+                let rhs_var = Self::extract_variable(rhs_inner, db);
+
+                if lhs_var == rhs_var {
+                    let op = binary_op.op(db);
+
+                    let simplified_op = match (op, lhs_inner.op(db), rhs_inner.op(db)) {
+                        (BinaryOperator::Or(_), BinaryOperator::EqEq(_), BinaryOperator::LT(_))
+                        | (BinaryOperator::Or(_), BinaryOperator::LT(_), BinaryOperator::EqEq(_))
+                        | (BinaryOperator::Or(_), BinaryOperator::LE(_), BinaryOperator::LT(_))
+                        | (BinaryOperator::Or(_), BinaryOperator::GE(_), BinaryOperator::GT(_)) => "<=",
```

Comment:
`x >= y || x > y` isn't equal to `x <= y`
```suggestion
                        | (BinaryOperator::Or(_), BinaryOperator::LE(_), BinaryOperator::LT(_)) => "<=",
```